src/main/java/com/github/hokkaydo/eplbot/module/code/RustCompiler.java

```diff
@@ -0,0 +1,106 @@
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+public class RustCompiler implements Runner{
+    private static final String CURRENT_DIR = System.getProperty("user.dir") + "\\src\\temp\\";
+    private static final String[] VALID_EXTENSIONS = {".exe", ".pdb", ".rs"};
+
+    private static final ScheduledExecutorService SCHEDULER = new ScheduledThreadPoolExecutor(1);
+    @Override
+    public String run(String input, Integer runTimeout) {
+        if (containsUnsafeKeywords(input)){
+            return "Compilation failed:\nCheck if 'std' or 'use' are used";
+        }
+        try {
+
+            File sourceFile = new File(CURRENT_DIR, "temp.rs");
+            sourceFile.deleteOnExit();
+            FileWriter writer = new FileWriter(sourceFile);
+            writer.write(input);
+            writer.close();
+
+            Process compile = new ProcessBuilder("rustc", sourceFile.getAbsolutePath()).directory(new File(CURRENT_DIR)).redirectErrorStream(true).start();
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+
+            int compileExitCode = compile.waitFor();
+            InputStream inputStream = compile.getInputStream();
+            byte[] buffer = new byte[1024];
+            int bytesRead;
+            while ((bytesRead = inputStream.read(buffer)) != -1) {
+                outputStream.write(buffer, 0, bytesRead);
+            }
+
+            if (compileExitCode != 0) {
+                deleteFiles();
+                return "Compilation failed:\n" + outputStream;
+            }
+            String executableFileName = sourceFile.getName().replace(".rs", "");
+            File executableFile = new File(CURRENT_DIR, executableFileName);
+
+            if (executableFile.setExecutable(true)) {
+                return "Couldn't make file executable";
+            }
+
+            Process run = new ProcessBuilder(new File(CURRENT_DIR, executableFileName).getAbsolutePath()).directory(new File(CURRENT_DIR)).redirectErrorStream(true).start();
+
+            ScheduledFuture<?> timeOut = SCHEDULER.schedule(() -> {}, runTimeout, TimeUnit.SECONDS);
+
+            outputStream.reset();
+            inputStream = run.getInputStream();
+            buffer = new byte[1024];
+            while ((bytesRead = inputStream.read(buffer)) != -1) {
+                outputStream.write(buffer, 0, bytesRead);
+            }
+            if (!timeOut.isDone()) {
+                deleteFiles();
+                return outputStream.toString();
+            }
+            deleteFiles();
+            String output = outputStream.toString().trim();
+
+            if (output.isEmpty()) {
+                return "Run failed: Timelimit exceeded "+ runTimeout +" s";
+            } else {
+                return "Run failed:\n" + output;
+            }
+        } catch (IOException | InterruptedException e) {
```

Comment:
The try-catch clause encloses too much. Additionally, call `Thread.currentThread().interrupt()` once you've processed it. It avoids loosing information that this thread has been interrupted